### DISCUSSION TITLE
Only Polygon GeoJSON supported

### DISCUSSION THREAD

**Discussion by ccall48** [uber/h3#783] [Q&A] [answered]
I'm fairly green with h3 so forgive me if this is ignorance on my behalf. I'm plotting a polygon of a landmass which i generated at geojson.io and trying to collect all the res8 hexagons within that geojson. below is a snippet of what i'm passing.

With js it manages to extract this data correctly with `geojson2h3.featureToH3Set(polygons, 8)`

```json
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "geometry": {
        "type": "Polygon",
        "coordinates": [[
            [142.2390605330495, -10.005513897126036], [141.16591324118713, -11.184615012913895]
            ....
        ]]
      }
    }
  ]
}
```
when I try to pass this with python using python, i get a `ValueError: Only Polygon GeoJSON supported`.
I have tried both `h3.polyfill_geojson(polygons, 8)` and `h3.polyfill(polygons, 8, geo_json_conformant=False)`.

Is the geojson structured correctly?

**Comment by nrabinowitz** [comment, marked as answer]
This is a better question for the `h3-py` repo, but I believe the issue here is that you can't pass a `FeatureCollection`, or any feature that's not a `Polygon`, to `polyfill_geojson`. You'll need to iterate through your features one by one, e.g.
```
out = set()
for feature in polygons.features:
  out.update(h3.polyfill_geojson(feature, 8))
```

**Comment by ccall48** [comment]
Thank you for your quick response.

Apologies if I have hit the wrong repo here, your advice is probably correct.
I just had another quick play and can confirm this does seem to work for returning all the hexes in a single feature.

```py
geo = geojson['features'][0]['geometry']

# return list of all hexes
print(h3.polyfill_geojson(geo, 8))

# print all hexes...
[print(x) for x in h3.polyfill_geojson(geo, 8)]
```
